libmscore/mmrest.cpp

```diff
@@ -0,0 +1,275 @@
+
+void MMRest::layout()
+{
+    m_number = measure()->mmRestCount();
+
+    for (Element* e : el()) {
+        e->layout();
+    }
+
+    if (score()->styleB(Sid::oldStyleMultiMeasureRests)) {
+        m_restSyms.clear();
+        m_symsWidth = 0;
+
+        int counter = m_number;
+        SymId sym;
+
+        while (counter >= 4) {
+            sym = SymId::restLonga;
+            m_restSyms.push_back(sym);
+            counter -= 4;
+            m_symsWidth += symBbox(sym).width();
+        }
+        while (counter >= 2) {
+            sym = SymId::restDoubleWhole;
+            m_restSyms.push_back(sym);
+            counter -= 2;
+            m_symsWidth += symBbox(sym).width();
+        }
+        if (counter == 1) {
+            m_isOddNumber = true;
+            // whole rest not added to m_restSyms, drawn separately due to need for vertical offset
+            m_symsWidth += symBbox(SymId::restWhole).width();
+            m_symsWidth += score()->scoreFont()->sym(sym).advance() * SPACING;
+        } else {
+            m_isOddNumber = false;
+        }
+        m_symsWidth += score()->scoreFont()->sym(sym).advance() * SPACING * (m_restSyms.size() - 1);
+    }
+
+    // set clickable area
+    if (score()->styleB(Sid::oldStyleMultiMeasureRests)) {
+        setbbox(QRectF((m_width - m_symsWidth) * .5, -1.0, m_symsWidth, 2.0));
+    } else { // H-bar}
+        qreal vStrokeHeight = score()->styleP(Sid::mmRestHBarVStrokeHeight);
+        setbbox(QRectF(0.0, -(vStrokeHeight * .5), m_width, vStrokeHeight));
+    }
+    addbbox(numberRect());
+}
+
+//---------------------------------------------------------
+//   numberRect
+///   returns the mmrest number's bounding rectangle
+//---------------------------------------------------------
+
+QRectF MMRest::numberRect() const
+{
+    std::vector<SymId>&& s = toTimeSigString(QString("%1").arg(m_number));
+
+    QRectF r = symBbox(s);
+    qreal x = (m_width - r.width()) * .5;
+    qreal y = m_numberPos * spatium() - staff()->height() * .5;
+
+    r.translate(QPointF(x, y));
+    return r;
+}
+
+//---------------------------------------------------------
+//   MMRest::write
+//---------------------------------------------------------
+
+void MMRest::write(XmlWriter& xml) const
+{
+    xml.stag("Rest");
+    ChordRest::writeProperties(xml);
+    el().write(xml);
+    xml.etag();
+}
+
+//---------------------------------------------------------
+//   propertyDefault
+//---------------------------------------------------------
+
+QVariant MMRest::propertyDefault(Pid propertyId) const
+{
+    switch (propertyId) {
+        case Pid::MMREST_NUMBER_POS:
+            return score()->styleV(Sid::mmRestNumberPos);
+        default:
+            return Rest::propertyDefault(propertyId);
+    }
+}
+
+//---------------------------------------------------------
+//   resetProperty
+//---------------------------------------------------------
+
+void MMRest::resetProperty(Pid propertyId)
```

Comment:
wasn't there already a parent class with similar code?

Reply:
Yes, you're right.